+ cd simulations
+ python simulate.py --program graph_coloring --sched 0 --no-sim 100000 --fault-interval 1 --graph-names star_graph_0_n7 --fault-prob 1.0 --simulation-type random
Locating Graph: "star_graph_0_n7".
Analysis graph: star_graph_0_n7 | program: graph_coloring | Simulation Type: random, Args: [] | No. of Simulations: 100000 | Scheduler: 0 | Mutual Exclusion: False | Fault Interval: 1
Total configs: 448.
Simulation environment: No. of Simulations: 100000 | Scheduler: CENTRAL | ME: False
Time taken: 9.0533s, Running simulation round: 50000
Time taken: 9.0378s, Running simulation round: 100000

Saving result at results/coloring/star_graph_0_n7__0__random_args___100000__False__1.csv
+ python simulate.py --program graph_coloring --sched 0 --no-sim 100000 --fault-interval 1 --graph-names star_graph_0_n7 --fault-prob 1.0 --simulation-type controlled_at_node_duong --controlled-at-node 0
Locating Graph: "star_graph_0_n7".
Analysis graph: star_graph_0_n7 | program: graph_coloring | Simulation Type: controlled_at_node_duong, Args: [0] | No. of Simulations: 100000 | Scheduler: 0 | Mutual Exclusion: False | Fault Interval: 1
Total configs: 448.
Simulation environment: No. of Simulations: 100000 | Scheduler: CENTRAL | ME: False
Time taken: 97.5227s, Running simulation round: 50000
Time taken: 96.5753s, Running simulation round: 100000

Saving result at results/coloring/star_graph_0_n7__0__controlled_at_node_duong_args_0__100000__False__1.csv
+ python simulate.py --program graph_coloring --sched 0 --no-sim 100000 --fault-interval 1 --graph-names star_graph_0_n7 --fault-prob 1.0 --simulation-type controlled_at_node_duong --controlled-at-node 4
Locating Graph: "star_graph_0_n7".
Analysis graph: star_graph_0_n7 | program: graph_coloring | Simulation Type: controlled_at_node_duong, Args: [4] | No. of Simulations: 100000 | Scheduler: 0 | Mutual Exclusion: False | Fault Interval: 1
Total configs: 448.
Simulation environment: No. of Simulations: 100000 | Scheduler: CENTRAL | ME: False
Time taken: 14.4089s, Running simulation round: 50000
Time taken: 14.3918s, Running simulation round: 100000

Saving result at results/coloring/star_graph_0_n7__0__controlled_at_node_duong_args_4__100000__False__1.csv
+ python simulate.py --program graph_coloring --sched 0 --no-sim 100000 --fault-interval 1 --graph-names star_graph_0_n7 --fault-prob 1.0 --simulation-type controlled_at_node_duong --controlled-at-node 5
Locating Graph: "star_graph_0_n7".
Analysis graph: star_graph_0_n7 | program: graph_coloring | Simulation Type: controlled_at_node_duong, Args: [5] | No. of Simulations: 100000 | Scheduler: 0 | Mutual Exclusion: False | Fault Interval: 1
Total configs: 448.
Simulation environment: No. of Simulations: 100000 | Scheduler: CENTRAL | ME: False
Time taken: 14.2095s, Running simulation round: 50000
Time taken: 14.3155s, Running simulation round: 100000

Saving result at results/coloring/star_graph_0_n7__0__controlled_at_node_duong_args_5__100000__False__1.csv
+ python simulate.py --program graph_coloring --sched 0 --no-sim 100000 --fault-interval 1 --graph-names star_graph_0_n7 --fault-prob 1.0 --simulation-type controlled_at_node_duong --controlled-at-node 9
Locating Graph: "star_graph_0_n7".
Analysis graph: star_graph_0_n7 | program: graph_coloring | Simulation Type: controlled_at_node_duong, Args: [9] | No. of Simulations: 100000 | Scheduler: 0 | Mutual Exclusion: False | Fault Interval: 1
Total configs: 448.
Simulation environment: No. of Simulations: 100000 | Scheduler: CENTRAL | ME: False
Traceback (most recent call last):
  File "/home/agaru/research/cvf-python-gnn/simulations/simulate.py", line 166, in <module>
    main(
    ~~~~^
        graph_name,
        ^^^^^^^^^^^
    ...<8 lines>...
        simulation_type_args,
        ^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/home/agaru/research/cvf-python-gnn/simulations/simulate.py", line 88, in main
    result = simulation.start_simulation(*simulation_type_args)
  File "/home/agaru/research/cvf-python-gnn/simulations/simulation.py", line 380, in start_simulation
    inner_results.append(self.run_simulations(state, *simulation_type_args))
                         ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/agaru/research/cvf-python-gnn/simulations/simulation.py", line 337, in run_simulations
    faulty_actions = faulty_action_generator(state, *extra_args, step)
  File "/home/agaru/research/cvf-python-gnn/simulations/simulation.py", line 322, in get_faulty_actions_controlled_at_node_duong
    faulty_actions = self.inject_least_fault_at_node(state, process)
  File "/home/agaru/research/cvf-python-gnn/simulations/simulation.py", line 201, in inject_least_fault_at_node
    p[process] = self.least_fault_weight
    ~^^^^^^^^^
IndexError: list assignment index out of range
